crates/core_simd/src/vector/mod.rs

```diff
@@ -0,0 +1,131 @@
+mod float;
+mod int;
+mod uint;
+
+pub use float::*;
+pub use int::*;
+pub use uint::*;
+
+// Vectors of pointers are not for public use at the current time.
+pub(crate) mod ptr;
+
+mod sealed {
+    pub trait Sealed {}
+}
+
+/// A representation of a vector as an "array" with indices, implementing
+/// operations applicable to any vector type based solely on "having lanes",
+/// and describing relationships between vector and scalar types.
+pub trait Vector: sealed::Sealed {
```

Comment:
So, I am fine with moving the trait into this module, but if we go to mod.rs it makes it harder to navigate because every mod.rs is the same as every other mod.rs in terms of terminal names (which isn't a problem now but may become one soon). I would prefer to retain the vector.rs style, even if it is just a "foyer" without much content of its own.